packages/objectloader2/src/types/types.ts

```diff
@@ -30,3 +32,19 @@ export function chunk<T>(array: T[], size: number): T[][] {
   }
   return result
 }
+
+export function isBase(maybeBase?: unknown): boolean {
+  return (
+    maybeBase !== null &&
+    typeof maybeBase === 'object' &&
```

Comment:
@adamhathcock I've just pushed in a commit adding [lodash](https://lodash.com/docs/4.17.15) as a dep. Its a common utility lib that will help you simplify some of these checks (e.g. isObjectLike, isString).

I noticed you already started creating your own versions for some of these in helpers, you can get rid of those and use lodash instead.

Just ensure you import from `#lodash` not `lodash` to ensure the proper ESM or CJS version is imported depending on the environment.

Reply:
Also, if you need to start getting creative with more complicated TS types, I suggest looking into https://github.com/sindresorhus/type-fest it has a lot of useful utility types

Reply:
I can do that....I was thinking I didn't want lodash anymore as a thing.  I'll clean up tomorrow